lib/fetchHttpClient.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import "isomorphic-fetch";
+import * as tough from "tough-cookie";
+import AbortController from "abort-controller";
+import FormData from "form-data";
+
+import { HttpClient } from "./httpClient";
+import { WebResource } from "./webResource";
+import { HttpOperationResponse } from "./httpOperationResponse";
+import { HttpHeaders } from "./httpHeaders";
+import { RestError } from "./restError";
+import { Readable, Transform } from "stream";
+
+interface FetchError extends Error {
+    code: string | undefined;
+    errno: string | undefined;
+    type: string | undefined;
```

Comment:
Nit: Instead of having undefined, you could also use the question mark pattern. ex: `code?: string`

Reply:
@daschult is `| undefined` kind of person so maybe we should discuss what's our official style ;)

Reply:
afaik they really are subtly different, one is an optional member while the other is required but you're allowed to pass `undefined` for it. So you should pick the one that has the behavior you want.